### PR TITLE
Add human relative time formatting utility

### PR DESCRIPTION
**Pull request by jacobtomlinson**
Added human relative time formatting utility `dask.utils.format_time_ago`.

```python
>>> from dask.utils import format_time_ago
>>> from datetime import datetime, timedelta

>>> now = datetime.now()
>>> format_time_ago(now)
'Just now'

>>> past = datetime.now() - timedelta(minutes=1)
>>> format_time_ago(past)
'1 minute ago'

>>> past = datetime.now() - timedelta(minutes=2)
>>> format_time_ago(past)
'2 minutes ago'

>>> past = datetime.now() - timedelta(hours=1)
>>> format_time_ago(past)
'1 hour ago'

>>> past = datetime.now() - timedelta(hours=6)
>>> format_time_ago(past)
'6 hours ago'

>>> past = datetime.now() - timedelta(days=1)
>>> format_time_ago(past)
'1 day ago'

>>> past = datetime.now() - timedelta(days=5)
>>> format_time_ago(past)
'5 days ago'

>>> past = datetime.now() - timedelta(days=8)
>>> format_time_ago(past)
'1 week ago'

>>> past = datetime.now() - timedelta(days=16)
>>> format_time_ago(past)
'2 weeks ago'

>>> past = datetime.now() - timedelta(days=190)
>>> format_time_ago(past)
'6 months ago'

>>> past = datetime.now() - timedelta(days=800)
>>> format_time_ago(past)
'2 years ago'
```

### CODE REVIEW

File: dask/utils.py [hunk 1]
```diff
@@ -1272,6 +1272,62 @@ def format_time(n):
     return "%.2f us" % (n * 1e6)
 
 
+def format_time_ago(n):
+    """Calculate a '3 hours ago' type string from a Python datetime.
+
+    >>> from datetime import datetime, timedelta
```
Comment (mrocklin) [inline dask/utils.py:1278]:
```suggestion

    Examples
    --------
    >>> from datetime import datetime, timedelta
```

Comment (mrocklin) [reply dask/utils.py:1278]: Can I also ask for a Parameters section with a type of `n` (or perhaps use Python type hints there?)

Maybe also add newlines in the example below in between the output of one section and the `past = ` line of the next?

**Review by mrocklin — COMMENTED** [review]
Minor docstring comments!

In general this seems like a nice thing to have.  Thanks for centralizing it here rather than elsewhere.

**Comment by mrocklin** [comment]
+1

**Comment by jacobtomlinson** [comment]
Thanks for the review @mrocklin. Updated!

**Pull request merged**